src/AddIns/Uno.WinUI.Graphics3DGL/GLCanvasElement.cs

```diff
@@ -169,8 +220,16 @@ private void OnClosed(object _, object __)
 	public void Invalidate() => NativeDispatcher.Main.Enqueue(Render, NativeDispatcherPriority.Idle);
 #endif
 
+	/// <summary>
+	/// Indicates whether this element was loaded successfully or not, including the OpenGL context creation and setup.
+	/// This property is only valid when the element is loaded. When the element is not loaded, the value will be null.
+	/// </summary>
+	public bool? LoadedSuccessfully { get; private set; }
```

Comment:
While this is useful to determine the state, it's best if we pdependency property instead. The developer using this will want to display something only if the control loads properly by using databinding. Also, for the name, `IsGLInitialized` would be best.

Reply:
We want this to be read-only, and unfortunately WinUI doesn't have read-only dependency properties, so I went with `INotifyPropertyChanged` instead. I don't like this approach either because now we're kind of forcing all `GLCanvasElement` subsclasses to also raise `PropertyChanged` on any exposed property. I don't think users will commonly want to add new properties to `GLCanvasElement` subclasses, but still.

Reply:
I really don't like this. IMHO, it's definitely not the user's problem if we expose something that is mutable, but is actually not supposed to be mutated. It's just bad design on WinUI's part. We should be making it statically-impossible to externally change this.

Reply:
Yes, but because the built-in WinUI controls do the same thing, I think it is better to be in-line with it and use dependency properties as well (as also implementing `INotifyPropertyChanged` on a control is not a ideal approach...). And if WinUI finally adds support for read-only DPs, this will stop being a problem

Theoretically we could also simulate read-only DP by adding a on dp changed handler and verifying if the value is being changed from internal code (by using some flag for example) - again not ideal, but it would prevent the user from changing it

Reply:
Fair enough.